From 2eced06d5f6f0c8b09d67466871ef3dfe92820b8 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Thu, 29 Nov 2018 15:57:05 +0800
Subject: [PATCH] usb: gadget: Kconfig: support Rockchip uvc

This patch adds support to use uvc for Rockchip ISP IQ tool
which needs large data buf of the uvc_request_data to send
vendor special uvc control data to uvc host.

Change-Id: I22b65c4f28d69b50f6bbb41124f3342b8c4c7c19
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/usb/gadget/Kconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/usb/gadget/Kconfig b/drivers/usb/gadget/Kconfig
index 5cf6802b02ae..3bcd31a49772 100644
--- a/drivers/usb/gadget/Kconfig
+++ b/drivers/usb/gadget/Kconfig
@@ -487,6 +487,15 @@ config USB_CONFIGFS_F_UVC
 	  device. It provides a userspace API to process UVC control requests
 	  and stream video data to the host.
 
+config USB_CONFIGFS_F_UVC_ROCKCHIP
+	bool "Rockchip USB Webcam function"
+	depends on USB_CONFIGFS_F_UVC && ARCH_ROCKCHIP
+	default y
+	help
+	  The Rockchip USB Webcam function is used for Rockchip ISP IQ tool.
+	  Enable this option will increase the UVC_MAX_REQUEST_SIZE from
+	  64 Bytes to 4100 Bytes.
+
 config USB_CONFIGFS_F_PRINTER
 	bool "Printer function"
 	select USB_F_PRINTER
-- 
2.35.3

